client/java/src/main/java/io/openlineage/client/dataset/Naming.java

```diff
@@ -0,0 +1,797 @@
+      return "kafka://" + bootstrapServerHost + ":" + port;
+    }
+
+    @Override
+    public String getName() {
+      return topic;
+    }
+  }
+
+  /** Naming implementation for Local File System. */
+  public static class LocalFileSystemNaming implements DatasetNaming {
+    private final String path;
+
+    public LocalFileSystemNaming(String path) {
+      this.path = checkArgumentNotEmpty(path, "path");
+    }
+
+    @Override
+    public String getNamespace() {
+      return "file";
+    }
+
+    @Override
+    public String getName() {
+      return path;
+    }
+  }
+
+  /** Naming implementation for Remote File System. */
+  public static class RemoteFileSystemNaming implements DatasetNaming {
+    private final String host;
+    private final String path;
+
+    public RemoteFileSystemNaming(String host, String path) {
+      this.host = checkArgumentNotEmpty(host, "host");
+      this.path = checkArgumentNotEmpty(path, "path");
+    }
+
+    @Override
+    public String getNamespace() {
+      return "file://" + host;
+    }
+
+    @Override
+    public String getName() {
+      return path;
+    }
+  }
+
+  /** Naming implementation for S3. */
+  public static class S3Naming implements DatasetNaming {
+    private final String bucketName;
+    private final String objectKey;
+
+    public S3Naming(String bucketName, String objectKey) {
+      this.bucketName = checkArgumentNotEmpty(bucketName, "bucketName");
+      this.objectKey = checkArgumentNotEmpty(objectKey, "objectKey");
+    }
+
+    @Override
+    public String getNamespace() {
+      return "s3://" + bucketName;
+    }
+
+    @Override
+    public String getName() {
+      return objectKey;
+    }
+  }
+
+  /** Naming implementation for WASBS. */
+  public static class WASBSNaming implements DatasetNaming {
+    private final String container_name;
+    private final String service_name;
+    private final String object_key;
+
+    public WASBSNaming(String container_name, String service_name, String object_key) {
+      this.container_name = checkArgumentNotEmpty(container_name, "container_name");
+      this.service_name = checkArgumentNotEmpty(service_name, "service_name");
+      this.object_key = checkArgumentNotEmpty(object_key, "object_key");
+    }
+
+    @Override
+    public String getNamespace() {
+      return "wasbs://" + container_name + "@" + service_name + ".dfs.core.windows.net";
+    }
+
+    @Override
+    public String getName() {
+      return object_key;
+    }
+  }
+
+  /**
+   * Naming implementation for Google Pub/Sub.
+   *
+   * <p>Namespace is fixed as {@code "pubsub"}. Name format depends on the resource type:
+   *
+   * <ul>
+   *   <li>{@code topic:{projectId}:{topicId}}
+   *   <li>{@code subscription:{projectId}:{subscriptionId}}
+   * </ul>
+   */
+  public static class PubSubNaming implements Naming.DatasetNaming {
+
+    /** Enum representing supported Pub/Sub resource types. */
+    public enum PubSubResourceType {
```

Comment:
How come this makes a difference? 
Isn't this breaking the lineage graph for jobs writing and reading from pubsub?